components/build/tekton-chains/tekton-chains-controller-shared-secret.yaml

```diff
@@ -0,0 +1,58 @@
+---
+# Mount the contents of the SharedSecret named redhat-appstudio-user-workload
+# on the Tekton Chains controller at $DOCKER_CONFIG/config.json. This makes
+# Chains consider those credentials when pushing content to an OCI repo.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tekton-chains-controller
+  namespace: tekton-chains
+spec:
+  template:
+    spec:
+      # Chains expects a file named config.json under the DOCKER_CONFIG dir.
+      # Since the underlying Secret defines the key .dockerconfigjson, that's
+      # also the name of the file that is mounted on the container.
+      #
+      # We could use subPath to "rename" the file. However, that causes two
+      # issues. First, updates to the Secret are not reflected on the container,
+      # see https://github.com/kubernetes/kubernetes/issues/50345. Second, if
+      # the underlying Secret does not exist, config.json is mounted as a dir
+      # instead of a file which breaks the k8schain library used by Tekton
+      # Chains.
+      #
+      # By using a init container, we can mount the volume as is and ensure that
+      # config.json (the symlink) is a file.
+      initContainers:
+      - name: link-shared-secret
+        image: registry.redhat.io/ubi8:latest
+        volumeMounts:
+        - name: default-push-secret
+          mountPath: /secret/default-push-secret
+        - name: docker-config
+          mountPath: /secret/docker-config
+        command:
+        - ln
+        - -s
+        - -f
+        - /secret/default-push-secret/.dockerconfigjson
+        - /secret/docker-config/config.json
```

Comment:
so do you know what `ls -la /secret/docker-config` looks like in the underlying Pod after the secret referenced by the `redhat-appstudio-user-workload` shared secret is created and the CSI driver attempts to store its contents ?

fwiw this sort of approach has never been tried before with shared resources

if I had cycles I'd try this out on the side but I don't know when in the short term I'll be able to

Reply:
I believe @lcarva told me in the hacbs weekly today that it worked (though I'm still mildly curious what the ls -la look like in the running pod if you `oc rsh` into it) ... which I think is also what he is conveying with https://github.com/redhat-appstudio/infra-deployments/pull/534#discussion_r932301954

Reply:
It's an empty directory if the `Secret` does not exist. As soon as it is created, it gets populated properly.

The initContainer workaround was suggested in the kubernetes issue linked in the comment.